jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/diagram/ProcessInstanceDiagramViewImpl.java

```diff
@@ -142,6 +159,71 @@ public void hideNodeActions() {
     @Override
     public void displayImage(final String svgContent) {
         diagram.displayImage(svgContent);
+        diagram.getElement().appendChild(nodeCounterView.getElement());
+    }
+
+    @Override
+    public void setNodeBadges(final Map<String, Long> badges) {
+        this.badges = badges;
+        renderBadges = true;
+    }
+
+    @Override
+    public void onShow() {
+        renderBadges();
+    }
+
+    protected void showHideBadges(){
+        final D3 d3 = D3.Builder.get();
+        final D3 nodes = d3.selectAll("#processDiagramDiv svg [jbpm-node-badge]");
+        nodes.attr("visibility", nodeCounterView.showBadges() ? "visible" : "hidden");
+    }
+
+    protected void renderBadges() {
+        if(renderBadges == false){
+            return;
+        }
+
+        final D3 d3 = D3.Builder.get();
+
+        final D3 svg = d3.select("#processDiagramDiv svg");
+        final D3.DOMRect svgRect = svg.node().getBoundingClientRect();
+        if(svgRect.getWidth() == 0 && svgRect.getHeight() == 0){
+            //SVG not visible
+            return;
+        }
+
+        badges.forEach((nodeId, count) -> {
+            final D3 node = d3.select("#processDiagramDiv svg [bpmn2nodeid=" + nodeId + "]");
+            D3.DOMRect bb = node.node().getBoundingClientRect();
+            final D3 group = node.append("g")
+                    .attr("transform", "translate( " + bb.getWidth() + ", " + bb.getHeight() + ")")
```

Comment:
@cristianonicolai I see the badge placement a bit weird over some components like start or split
![processinstancebadge](https://user-images.githubusercontent.com/2443754/51759020-84cb2f80-20c7-11e9-8af6-6549332bc5ef.png)
I tried to see if bb.getWidth()/2 but not sure if it looks better or if it is further of the UX proposed design, what do you think?
![processinstancebadge2](https://user-images.githubusercontent.com/2443754/51759436-83e6cd80-20c8-11e9-8903-ca5c70792937.png)